### PR TITLE
Extra requests for Conan package uploads

### PR DESCRIPTION
**Pull request by ChGen**
Part of #2 - implements extra requests, required for Conan package uploads. 1 new commit.

### CODE REVIEW

**Comment by ChGen** [comment]
@olenagerasimova , please review.

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@ChGen thanks, one comment. And again - there are no unit tests

**Comment by ChGen** [comment]
please, review.As for tests, here's mainly http requests for Conan client, so there will be integration tests later. Probably as part of the main Artipie project.

**Comment by ChGen** [comment]
@olenagerasimova, please, review. 
As for tests, here's mainly http requests for Conan client, so there will be integration tests later. I think they will be part of the main Artipie project repo.

File: src/main/java/com/artipie/conan/http/BaseConanSlice.java [hunk 1]
```diff
@@ -134,11 +138,40 @@ protected Storage getStorage() {
...
+                if (exist) {
+                    result = this.storage.value(key).thenCompose(
+                        content -> {
+                            final ContentDigest dgt = new ContentDigest(
+                                content, () -> {
+                                try {
+                                    return MessageDigest.getInstance("MD5");
+                                } catch (final NoSuchAlgorithmException ex) {
+                                    throw new ArtipieIOException(ex);
+                                }
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/conan/http/BaseConanSlice.java:159]: there also is a `Digests` enum in `asto`, use `MD5` constant

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
One more suggestion, also, I did not ask about ITs, I've asked about unit tests. Each adapter have unit test for each slice, there are not many exceptions.

File: src/main/java/com/artipie/conan/http/BaseConanSlice.java [hunk 2]
```diff
@@ -134,11 +136,33 @@ protected Storage getStorage() {
...
+    protected CompletableFuture<String> generateMDhash(final Key key) {
+        return this.storage.exists(key).thenCompose(
+            exist -> {
+                final CompletableFuture<String> result;
+                if (exist) {
+                    result = this.storage.value(key).thenCompose(
+                        content -> {
+                            final ContentDigest dgt = new ContentDigest(content, Digests.MD5);
+                            return dgt.hex();
+                        });
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/conan/http/BaseConanSlice.java:153]: the expression can be simplified to 
```
result = this.storage.value(key).thenCompose(contente -> new ContentDigest(content, Digests.MD5).hex())
```

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@ChGen one more suggestion

**Comment by ChGen** [comment]
@olenagerasimova , please review.

**Review by olenagerasimova — APPROVED** [review]
@ChGen thanks

**Pull request merged**